### PR TITLE
Framework introduction

### PR DESCRIPTION
**Pull request by henriknorin-ftrack**
Resolves : 

* CLICKUP-3jv1140

This PR has been tested on :

- [X] MacOs.

## Changes

Finished the framework introduction **and developing** sub section.

## Test

Build and validate the documentation: 

`python setup.py build_sphinx`

### CODE REVIEW

File: doc/developing/build.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+****************
+Build and deploy
+****************
+
+.. highlight:: bash
+
+TBC
+
+Building and deploying Connect centrally
```
Comment (hdd) [inline doc/developing/build.rst:14]: @henriknorin-ftrack 

As we have it, might be better to reference the existing connect's documentation.

File: doc/developing/download.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+Download
+********
+
+.. highlight:: bash
+
+The framework is 100% open source and accessible on GitHub.
+
+Git clone base repositories::
+
+    gh repo clone ftrackhq/ftrack-connect-pipeline
```
Comment (hdd) [inline doc/developing/download.rst:16]: @henriknorin-ftrack 
git clone is better than gh as it is gihub specific only and we want to be as generic as possible.

File: doc/glossary.rst [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        target platform. typically Windows, Mac OS and Linux. If supplies a default
+        Python runtime for running a Connect as an executable, compiled using cx_freeze.
+        Source code can be found here: https://github.com/ftrackhq/ftrack-connect-package.git
+
+    DCC
+        Digital Content Creation tool, e.g. Maya, 3D Studio Max, Unreal, Blender and
+        so on. Each DCC application has an associated framework plugin, the current ones
+        defined are:
+
+         * Maya; https://github.com/ftrackhq/ftrack-connect-pipeline-maya.git
```
Comment (hdd) [inline doc/glossary.rst:43]: @henriknorin-ftrack no need to have repositories defined here, you already have a list of applications.

File: doc/installing.rst [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+.. _installing:
+
+**********
+Installing
+**********
+
+.. highlight:: bash
+
+Install the DCC framework plugins through the plugin manager within :term:`Connect` 2.
```
Comment (hdd) [inline doc/installing.rst:12]: @henriknorin-ftrack you can link the plugin manager documentation to close the loop
https://github.com/ftrackhq/ftrack-connect-plugin-manager

**Comment by hdd** [comment]
@henriknorin-ftrack 

I'd like to see , where is possible and make sense, the use of intersphinx to wire Connect, Application launcher, plugin manager, publisher etc....

this would make this guide more useful and handier to use.

File: doc/introduction/connect.rst [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+:term:`Connect` is the ftrack desktop GUI application enabling DCC application launch
+and publishing. Connect locates the framework plugins, and together with the
+:term:`Application launcher` plugin, enables discover and launch of DCC applications.
+
+The framework modules are connect plugins, which means that need to implement a launch
+hook containing the logic to discover and launch the DCC application.
+
+
+Application launcher
```
Comment (hdd) [inline doc/introduction/connect.rst:18]: @henriknorin-ftrack there's no need here to have reference to the configs, we have the applicatino launcher documentation.

File: doc/introduction/framework/core.rst [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+other modules rely.
+
+It is in the core were all the interaction with the underlaying :term:`host type` is
+performed, except when it comes to bootstrap of the DCC.
+
+The core is depending on the :term:`definition` module to be present.
+
+The module composes four major components:
+
+ * The Host
```
Comment (hdd) [inline doc/introduction/framework/core.rst:24]: @henriknorin-ftrack 

link these entries to the page subsections so they become clickable

File: doc/introduction/framework/core.rst [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+ * Opener
+
+Logs
+----
+
+Clients sends notifications to the host which is stored in an internal SQLite database
+valid during the session.
+
+
+Customisation notes
```
Comment (hdd) [inline doc/introduction/framework/core.rst:74]: @henriknorin-ftrack 

Customization notes, guess should be in the development part ?

File: doc/index.rst [hunk 8]
```diff
@@ -3,15 +3,55 @@
...
+
+This document is intended to bring a deeper understanding of how the new Connect 2
+:term:`framework` works under the hood, enabling developers to customize it to their
+particular needs.
+
+For a general introduction on how to install and use ftrack Connect 2 out of the box,
+please head over here: https://www.ftrack.com/en/portfolio/connect
+
+The documentation is divided into two main parts, the first part is where we describe
+the new Connect 2 framework and the second part is where we customise it to fit our needs.
```
Comment (lluisFtrack) [inline doc/index.rst:28]: Are we shure we want to name it the new Connect2 framework as it was the same thing? or just new framework as it's not strictly related with connect?

**Comment by hdd** [comment]
@henriknorin-ftrack 

some parts could be better rendered with something like [dot/graphviz](https://www.sphinx-doc.org/en/master/usage/extensions/graphviz.html)

File: doc/introduction/overview.rst [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+ * :term:`Python`, specifically the ftrack Python API.
+ * Source code management, specifically Git.
+ * :term:`DCC` applications, i.e. Maya, Nuke.
+ * ftrack :term:`Connect` 2 Package, installation - app launch - publish and load.
+
+
+Key design elements and tradeoffs
+=================================
+
+Connect 2 comes with a new framework, designed to be a starting point for developers
```
Comment (lluisFtrack) [inline doc/introduction/overview.rst:25]: The pre-build of Connect 2 comes with the new framework or they have to download it throw the Plugin manager?

File: doc/glossary.rst [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+    Python
+        A programming language that lets you work more quickly and integrate
+        your systems more effectively. Often used in creative industries. Visit
+        the language website at http://www.python.org
+
+    Qt
+        The default UI framework utilised by the Framework, through PySide and
+        the Qt.py Python binding module. The corresonding Framework module containing
+        UI bindings is named ftrack-connect-pipeline-qt and its source code can be
+        found here: https://github.com/ftrackhq/ftrack-connect-pipeline-qt.git
```
Comment (hdd) [inline doc/glossary.rst:102]: @henriknorin-ftrack no need to reference the repo here.
One thing is the glossary for the technology ,another is how we use it.

File: doc/glossary.rst [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+
+        Further resources:
+         * Qt; https://www.qt.io/
```
Comment (hdd) [inline doc/glossary.rst:105]: @henriknorin-ftrack 

you can lave just the reference to qt.io in case

**Review by lluisFtrack — COMMENTED** [review]
Looks good to me, just a couple of comments added.

File: doc/introduction/framework/dcc.rst [hunk 12]
```diff
@@ -0,0 +1,52 @@
+..
+    :copyright: Copyright (c) 2022 ftrack
+
+.. _introduction/framework/dcc:
+
+*********************
+DCC integration layer
+*********************
+
+Repository: https://github.com/ftrackhq/ftrack-connect-pipeline-<host_type>.git
```
Comment (hdd) [inline doc/introduction/framework/dcc.rst:10]: @henriknorin-ftrack maybe host_type is something we can try to reference and link from the glossary ?

File: doc/release/migrating_from_legacy_connect_framework.rst [hunk 13]
```diff
@@ -0,0 +1,30 @@
+..
+    :copyright: Copyright (c) 2022 ftrack
+
+.. _release/migrating_from_legacy_connect_framework:
+
+************************************
+Migrating from old Connect framework
```
Comment (hdd) [inline doc/release/migrating_from_legacy_connect_framework.rst:7]: @henriknorin-ftrack 

old code was under the name:

**ftrack connectors** , users might not know it as **legacy framework** .

File: doc/tutorial/introduction.rst [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+    provide separate builds and configurations per project or context in general.
+    This is subject to be improved with the next major release of the framework.
+
+
+Checkout documentation code
+---------------------------
+
+The source code from this documentation, as fully working examples, can be found
+in the *resource* directory.
+
```
Comment (hdd) [inline doc/tutorial/introduction.rst:30]: @henriknorin-ftrack add how to build the docs while here ? :)

**Review by hdd — CHANGES_REQUESTED** [review]

File: doc/introduction/overview.rst (hunk 9 above)
Comment (henriknorin-ftrack) [reply doc/introduction/overview.rst:25]: Removed Connect 2 here aswell

File: doc/installing.rst (hunk 4 above)
Comment (henriknorin-ftrack) [reply doc/installing.rst:12]: Adding glossary for _plugin manager_, with intersphinx link

File: doc/tutorial/introduction.rst (hunk 14 above)
Comment (henriknorin-ftrack) [reply doc/tutorial/introduction.rst:30]: Have added this to the developer section.

**Review by hdd — APPROVED** [review]

**Pull request merged**